task_02/xieyu_wang/graphic.cpp

```diff
@@ -0,0 +1,114 @@
+#include <array>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+constexpr int MAX_ELEMENT = 18;
+struct Data {
+  int from;
+  int edge;
+  int to;
+};
```

Comment:
Why are you combining everything into one struct? You shouldn't try to generalise. Here, it is actually opposite, it makes sense to define different structures for different entities, as they are not similar, they are different.
You should have both. If you have been writing JAVA, it is actually how you will do it there:
```suggestion
class Vertex {
 public:
  const int id = 0;
  
  Vertex(int _id) : id(_id) {}
  
  void add_edge_id(int id) {
    // validation that id has not been added already
    edge_ids_.push_back(id);
  }
  
 private:
  std::vector<int> edge_ids_;
};

class Edge {
 public:
  const int id = 0;
  const int from_vertex_id = 0;
  const int from_edge_id = 0;
  
  Edge(int _id, int _from_vertex_id, int _to_vertex_id) : id(_id), from_vertex_id(_from_vertex_id), to_vertex_id(_to_vertex_id) {}
};
```